pkg/services/sqlstore/datasource.go

```diff
@@ -289,6 +290,28 @@ func (ss *SQLStore) UpdateDataSource(ctx context.Context, cmd *datasources.Updat
 	})
 }
 
+func (ss *SQLStore) UpdateCorrelations(ctx context.Context, cmd *datasources.UpdateCorrelationsCommand) error {
+	return ss.WithTransactionalDbSession(ctx, func(sess *DBSession) error {
+		ds := &datasources.DataSource{
+			Correlations: cmd.Correlations,
+			Updated:      time.Now(),
+		}
+
+		// TODO: should we also check for version here instead (like UpdateDatasource)?
```

Comment:
`version` is used in updateDatasource to handle concurrent edits, shall we do the same here?

Reply:
yes, my opinion is that it should behave the same.

given here the datasource table record is updated, we need also to increment the version when correlations are updated ([same as we do when updating the datasource](https://github.com/grafana/grafana/blob/0c04fe16aab15e6aab6233d42a9a1e123b56458b/pkg/services/sqlstore/datasource.go#L245)).